source/Applications/Advanced/MaskPointCloud/MaskPointCloud.cpp

```diff
@@ -0,0 +1,224 @@
+        for(size_t i = 0; i < pointCloud.height; i++)
+        {
+            for(size_t j = 0; j < pointCloud.width; j++)
+            {
+                if(std::isnan(pointCloud(j, i).z))
+                {
+                    z.at<uchar>(i, j) = 0;
+                }
+                else
+                {
+                    z.at<uchar>(i, j) =
+                        static_cast<unsigned char>((255.0F * (pointCloud(j, i).z - zMin) / (zMax - zMin)));
+                }
+            }
+        }
+
+        // Applying color map
+        cv::Mat zJetColorMap;
+        cv::applyColorMap(z, zJetColorMap, cv::COLORMAP_JET);
+
+        // Setting invalid points (nan) to black
+        for(size_t i = 0; i < pointCloud.height; i++)
+        {
+            for(size_t j = 0; j < pointCloud.width; j++)
+            {
+                if(std::isnan(pointCloud(j, i).z))
+                {
+                    auto &zRGB = zJetColorMap.at<cv::Vec3b>(i, j);
+                    zRGB[0] = 0;
+                    zRGB[1] = 0;
+                    zRGB[2] = 0;
+                }
+            }
+        }
+        return zJetColorMap;
+    }
+
+    pcl::PointCloud<pcl::PointXYZRGB> maskPointCloud(const Zivid::PointCloud &pointCloud, const cv::Mat &mask)
+    {
+        const auto data = pointCloud.copyPointsXYZColorsRGBA();
+        const int height = data.height();
+        const int width = data.width();
+
+        // Creating point cloud structure
+        pcl::PointCloud<pcl::PointXYZRGB> maskedPointCloud(width, height);
+        maskedPointCloud.is_dense = false;
+        maskedPointCloud.points.resize(height * width);
+
+        // Copying data points within the mask. Rest is set NaN
+        for(int i = 0; i < height; i++)
+        {
+            for(int j = 0; j < width; j++)
+            {
+                if(mask.at<uint8_t>(i, j) > 0)
+                {
+                    maskedPointCloud(j, i).r = data(i, j).color.r;
+                    maskedPointCloud(j, i).g = data(i, j).color.g;
+                    maskedPointCloud(j, i).b = data(i, j).color.b;
+                    maskedPointCloud(j, i).x = data(i, j).point.x;
+                    maskedPointCloud(j, i).y = data(i, j).point.y;
+                    maskedPointCloud(j, i).z = data(i, j).point.z;
+                }
+                else
+                {
+                    maskedPointCloud(j, i).r = 0;
+                    maskedPointCloud(j, i).g = 0;
+                    maskedPointCloud(j, i).b = 0;
+                    maskedPointCloud(j, i).x = NAN;
+                    maskedPointCloud(j, i).y = NAN;
+                    maskedPointCloud(j, i).z = NAN;
+                }
+            }
+        }
+
+        return maskedPointCloud;
+    }
+
+    void visualizeDepthMap(const pcl::PointCloud<pcl::PointXYZRGB> &pointCloud)
+    {
+        std::cout << "Converting to Depth map in OpenCV format" << std::endl;
```

Comment:
In some functions you print out (std::cout) and in some you comment only. Pick one?